florist/api/server.py

```diff
@@ -1,8 +1,31 @@
 """FLorist server FastAPI endpoints and routes."""
+from contextlib import asynccontextmanager
+from typing import Any, AsyncGenerator
+
 from fastapi import FastAPI
+from pymongo import MongoClient
 
+from florist.api.routes.server.job import router as job_router
 from florist.api.routes.server.training import router as training_router
 
 
-app = FastAPI()
+MONGODB_URI = "mongodb://localhost:27017/"
+DATABASE_NAME = "florist-server"
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI) -> AsyncGenerator[Any, Any]:
+    """Set up function for app startup and shutdown."""
+    # Set up mongodb
+    app.mongodb_client = MongoClient(MONGODB_URI)  # type: ignore[attr-defined]
```

Comment:
Since FastAPI is async and [MongoClient is not](https://www.mongodb.com/docs/drivers/pymongo/#introduction), would it be good to use asynchronous python driver like [Motor](https://motor.readthedocs.io/en/stable/index.html). Although maybe this overkill at this point in the project. Just wanted to mention in case you think it makes sense :)

Reply:
I think it's easy enough to switch. Thanks for pointing that out :)